### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by Paul-js-hub**
### In this project I:
- Set up associations between classes and objects.
- Added students to the classroom.
- Added rentals to the book and person.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require_relative 'rental'
+class Book
+  attr_accessor :title, author, rentals
+
```
Comment (tresorsawasawa) [inline book.rb]: - [ ] Kindly make sure you correct the typing errors on the `attr_accessor` values to make it professional.
```ruby
# Bad
attr_accessor :title, author, rentals

# Good
attr_accessor :title, :author, :rentals
```

File: person.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
 require_relative 'nameable_interface'
+require_relative 'rental'
+
 class Person < Nameable
   attr_reader :id
-  attr_accessor :name, :age, nameable
+  attr_accessor :name, :age
 
   def initialize(name, age = 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
```
Comment (tresorsawasawa) [inline person.rb]: - [ ] Kindly make sure `Getters` and `Setters` for `@rentals` are present as required for the project **i.e** `:rentals` should be added on the `attr_accessor` values and be added in the constructor `initialize` method(e.g.: ``@rentals = []``)

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @Paul-js-hub 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShadyShawkat — APPROVED** [review]
## Hi @Paul-js-hub, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**